pkg/core/native/native_neo.go

```diff
@@ -442,6 +442,15 @@ func (n *NEO) distributeGas(ic *interop.Context, h util.Uint160, acc *state.NEOB
 		return err
 	}
 	acc.BalanceHeight = ic.Block.Index
+
+	// Must store acc before GAS distribution to fix acc's BalanceHeight value in the storage for
+	// further acc's queries from `onNEP17Payment` if so.
+	key := makeAccountKey(h)
+	err = ic.DAO.PutStorageItem(n.ID, key, acc.Bytes())
```

Comment:
What's interesting is that the balance value is actually going to be the old one.

Reply:
One more interesting thing is that [LazyCoin contract](https://github.com/vang1ong7ang/Tanya/blob/main/PoC/InsufficientFunds.cs#L29) should own some NEO for a time period to trigger GAS mint inside `onNEP17Payment`. But the whole NEO balance is transferred to the owner each time onNEP17Payment is called.